submissions/5mountains/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,102 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Resident {
+  constructor(species, name, gender, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.friends = friends;
+    this.saying = saying;
+  }
+  toString() {
+    return `Wonderful creature - <b>${this.species}</b>, 
+      whose name is <b>${this.name}</b>, 
+      the gender is <b>${this.gender}</b>! 
+      Usually the greeting is <b>${this.saying}</b> 
+      The friends of this inhabitant are <b>${this.friends.join(', ')}</b>. `;
+  }
+}
+
+class Human extends Resident {
+  constructor(name, gender, saying, friends, legs = 2, hands = 2) {
+    super('human', name, gender, saying, friends);
+    this.legs = legs;
+    this.hands = hands;
+  }
+  toString() {
+    return (
+      super.toString() +
+      `By the wat, this member is owner of <b>${this.legs} legs</b> and <b>${this.hands} hands</b>!`
+    );
+  }
+}
+
+class Animal extends Resident {
+  constructor(species, name, gender, saying, friends, paws = 4) {
+    super(species, name, gender, saying, friends);
+    this.paws = paws;
+  }
+  toString() {
+    return (
+      super.toString() +
+      `By the wat, this member is owner of <By>${this.paws} paws</By>!`
+    );
+  }
+}
+
+class Dog extends Animal {
+  constructor(name, gender, saying, friends) {
+    super('dog', name, gender, saying, friends);
+  }
+}
+
+class Cat extends Animal {
+  constructor(name, gender, saying, friends) {
+    super('cat', name, gender, saying, friends);
+  }
+}
+
+class CatWoman extends Human {
+  constructor(name, saying, friends = ['Tom', 'Felix']) {
+    super(name, 'female', saying, friends);
+    this.saying = CatWoman.saying();

```

Comment:
`saying` received by the class constructor is sent over its super and then overwritten.
1. Do we need to CatWoman constructor to expect any saying at all? Imagine a peer dev who does `new CatWoman('Kate', 'Meohow!')` and this brings no effect.
2. If super assign saying why not sending value for it instead of overwriting the property?

`callFunction(5, varName, someFunction())` is pretty valid syntax, `callFunction` will receive `5`, value of `varName` and whatever `someFunction()` returns.

Reply:
Aha, I think I got it, in this case, I should totally remove this.saying property inside CatWoman class and send the way of calling this static method into parent constructor via super() ? :)

Reply:
try it

Reply:
It's alive! I feel like a mad scientist!